### ISSUE TITLE
Is this intended to work with clangd or just libclang?

### ISSUE THREAD

**Issue by devjgm** [grailbio/bazel-compilation-database#37]
I'm trying to use `ycm` (in vim) with `clangd` (not libclang), and I'm trying to use this project to generate my compile_commands.json file for me. Specifically, I'm trying to use this as a vim plugin that uses bazel aspects to automatically generate the compile_commands for clangd.

I cannot get this to work. Before I spend more time debugging this, is this intended to work with clangd too?

**Comment by siddharthab** [comment]
I have never tested with clangd explicitly.

**Comment by devjgm** [comment]
Thanks.

I *think* `clangd` needs an explicit compile_commands.json file. So I also tried to use the `generate.sh` script to generate one, and that did not work. Here's the error I got when using the bazel build from the cctz project [a fairly simple bazel build]).

```shell
$ git clone git@github.com:devjgm/cctz cctz
[ ... ]
$ cd cctz
$ ~/.vim/bundle/bazel-compilation-database/generate.sh
[...]
ERROR: Label '//home/jgm/.vim/bundle/bazel-compilation-database:aspects.bzl' is invalid because 'home/jgm/.vim/bundle/bazel-compilation-database' is not a package; perhaps you meant to put the colon here: '//:home/jgm/.vim/bundle/bazel-compilation-database/aspects.bzl'?
ERROR: Analysis of aspect '//home/jgm/.vim/bundle/bazel-compilation-database:aspects.bzl%compilation_database_aspect of //:time_zone_lookup_test' failed; build aborted: Label '//home/jgm/.vim/bundle/bazel-compilation-database:aspects.bzl' is invalid because 'home/jgm/.vim/bundle/bazel-compilation-database' is not a package; perhaps you meant to put the colon here: '//:home/jgm/.vim/bundle/bazel-compilation-database/aspects.bzl'?
INFO: Elapsed time: 0.362s
INFO: 0 processes.
$
```

I'm using bazel 1.0.0.

Any thoughts? Should that use of `generate.sh` work?

**Comment by siddharthab** [comment]
Thank you for reporting this issue. The current generate.sh was written sometime ago and was not using some newer bazel features, hence it had limitations that it needed to be used when the file itself is present in the repo.

I have now made some code changes and also improved the documentation. Getting the latest version will make your command work.

Fixed by 6347f8b21c9a6c76367b63b3e689a8cd16a48340.
